common/src/Model/ModelTypes.h

```diff
@@ -26,6 +26,7 @@
 
 #include <vecmath/polygon.h>
 
+// FIXME: this header must not pull in std headers
```

Comment:
Yeah. I was looking at this a few weeks ago. I can see 3 options for the collections:

- We could just drop these aliases, search/replace "AttributableNodeList" with "std::vector<AttributableNode*>", etc. project wide
- Duplicate exactly the group of aliases that are needed in each header that uses them
- Have ModelLists.h (includes std::vector) ModelSets.h (includes set) ModelMaps.h, and include the relevant ones.

Reply:
I'm not sure. For simplicity and readability, at this point I am leaning towards option one. The only reason to have these particular aliases is for making changes easier (can substitute std::list for std::vector in just one place). I initially also thought this would improve readability, but it does the opposite, because the alias doesn't really add any more semantics, it takes away from the original name (is it a std::list? a std::vector?). So I think I'd be going with option 1 at this point.

Reply:
Yeah agreed